artifact.go

```diff
@@ -8,51 +8,42 @@ import (
 	"mime"
 	"net/http"
 	"path/filepath"
+	"time"
 
 	"github.com/gin-gonic/gin"
 	"github.com/iver-wharf/wharf-core/pkg/ginutil"
 	"gorm.io/gorm"
 )
 
+const (
+	buildIDParamName    = "buildid"
+	artifactIDParamName = "artifactId"
+)
+
 type ArtifactModule struct {
 	Database *gorm.DB
 }
 
-type TestRun struct {
-	XMLName       xml.Name      `xml:"TestRun"`
-	ResultSummary ResultSummary `xml:"ResultSummary"`
-}
-
-type ResultSummary struct {
-	XMLName  xml.Name `xml:"ResultSummary"`
-	Counters Counters `xml:"Counters"`
-}
-
-type Counters struct {
-	XMLName xml.Name `xml:"Counters"`
-	Passed  int      `xml:"passed,attr"`
-	Failed  int      `xml:"failed,attr"`
-}
-
-type TestStatus string
-
-const (
-	TestStatusSuccess TestStatus = "Success"
-	TestStatusFailed  TestStatus = "Failed"
-	TestStatusNoTests TestStatus = "No tests"
-)
-
-type TestsResults struct {
-	Passed int        `json:"passed"`
-	Failed int        `json:"failed"`
-	Status TestStatus `json:"status" enums:"Success,Failed,No tests"`
+// SummaryOfTestResultSummaries contains data about several test result files.
+type SummaryOfTestResultSummaries struct {
+	BuildID   uint                `json:"buildId"`
+	Total     uint                `json:"total"`
+	Failed    uint                `json:"failed"`
+	Passed    uint                `json:"passed"`
+	Skipped   uint                `json:"skipped"`
+	Summaries []TestResultSummary `json:"summaries"`
```

Comment:
That name though...

Could perhaps be enough with only 1 "Summary" in the name.

What about this?:

```go
// TestResultListSummary contains data about several test result files.
type TestResultListSummary struct {
	BuildID uint                `json:"buildId"`
	Total   uint                `json:"total"`
	Failed  uint                `json:"failed"`
	Passed  uint                `json:"passed"`
	Skipped uint                `json:"skipped"`
	Results []TestResultSummary `json:"results"`
}
```

or maybe:

```go
// BuildTestResultSummary contains summary of tests for a given build.
type BuildTestResultSummary struct {
	BuildID uint                `json:"buildId"`
	Total   uint                `json:"total"`
	Failed  uint                `json:"failed"`
	Passed  uint                `json:"passed"`
	Skipped uint                `json:"skipped"`
	Results []TestResultSummary `json:"results"`
}
```

Reply:
Decided to go with `TestResultListSummary`.